deploy/crds/litmuschaos_v1alpha1_chaosexperiment_cr.yaml

```diff
@@ -1,7 +1,43 @@
-apiVersion: litmuschaos.io/v1alpha1
+---
+## An experiment is the definition of a chaos test and is listed as an item 
+## in the chaos engine to be run against a given app. Consists of reference
+## to chaosGraph (which has chaos param details and executor spec) and also
+## component information for the Object-Under-Test. The results upon running
+## this experiment is relayed back to the chaos engine which invoked it.
+
+## Experiments can be pulled from litmus charts and are versioned based on 
+## chart.
+
+## Charts should (not?) be auto-upgraded
+
+apiVersion: litmus.io/v1alpha1
 kind: ChaosExperiment
 metadata:
-  name: example-chaosexperiment
+
+  ## Eventually launched chaos litmusbook/job will bear <name>-<hash>
+  name: disappearing-pods
```

Comment:
IMO this should convey what it will try to do. In other words, it should convey the exact chaos experiment that will be done.

Reply:
Can `ChaosExperiment` & `ChaosTemplate/ChaosGraph` have same name?
If above is a good thing to do, then we can remove the entire mapping to `ChaosTemplate/ChaosGraph` from within `ChaosExperiment` entirely.

AFAIK ChaosExperiment will do only one specific task. ChaosGraph has a direct relation with underlying infrastructure to execute this specific task. e.g. spawn a k8s job that deletes a pod.

Reply:
We might need to rethink if infrastructure should be abstracted as well. e.g. killing a pod via a k8s job might be different from killing a pod via some api service or other chaos tooling.

Reply:
- Agree on name changes, will do that. 

- Very nice! Will think on addition to chaosGraph spec to package-in the executor nature (via traditional k8s litmus job, some other api - maybe directly reuse chaostoolkit api for example etc.,) 

- The experiment-graph/template was originally conceived in order to keep the spec of a "test"/"chaos experiment" homogeneous and common irrespective of what it does underneath. This interface would then reference low-level specs which can change greatly on a case-by-case basis. Now there are two developments which make us rethink this approach: 

  a/ Emergence on chaosEngine which is now the app/user-facing resource - which will be homogeneous. 

  b/ Introduction of `spec.definition` in the chaosGraph which packages all heterogeneous chaos parameters as ENV - which satisfies in a way the strict-definition/uniformity requirements of chaosGraph. 

  Now, even w/ (a) & (b), are there are some situations which necessitates two separate CRs ? I can think of this future case: An experiment might be the result of a (chaosjob-1) + (chaosjob-2) + (chaosjob-3) executed in that order to simulate a "multi-component" failure or an "issue-build-up" ? It can be a developer discretion also, to define it that way (This requirement varies slightly from the engine listing <exp1, exp2, exp3 by rank> - this is more of a batch run of chaos like diagnostics... the case we are discussing is the experiment-itself which is a case of more than 1 chaos event/action.) This thinking violates the current approach of 1-1 mapping, and may need a corresponding change to the spec  - but is an important consideration nonetheless.

Reply:
Current decision on the 3rd point based on discussions with @AmitKumarDas : 

- Multi-chaos experiments is definitely a possibility - but we currently feel these are best handled by the executors. Exposing programming constructs into the YAMLs may not be wise at this point.
- However, we will still go ahead with both the CRs to cover for cases where a user-defined chaosGraph is added and experiment can refer to it while maintaining the components abstraction. Future ones might come up. Moreover, there is no controller mapped against the resource anyway - it is only to hold data.
- We will have same names for chaosExperiment and chaosGraph to enable simple identification. User can specify different on a need basis.
- It will be of great value to add a "description" attribute to the experiment spec to hold information about what it does. This will be very useful for developers and help abstract-away lower-level chaos details (also another reason to maintain a separate CR).